storage/aptosdb/src/pruner/ledger_store/ledger_store_pruner.rs

```diff
@@ -2,181 +2,154 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{
-    db_metadata::DbMetadataSchema,
+    ledger_db::LedgerDb,
     metrics::PRUNER_VERSIONS,
     pruner::{
         db_pruner::DBPruner,
         db_sub_pruner::DBSubPruner,
         event_store::event_store_pruner::EventStorePruner,
-        ledger_store::version_data_pruner::VersionDataPruner,
+        ledger_store::ledger_metadata_pruner::LedgerMetadataPruner,
         transaction_store::{
-            transaction_store_pruner::TransactionStorePruner, write_set_pruner::WriteSetPruner,
+            transaction_accumulator_pruner::TransactionAccumulatorPruner,
+            transaction_info_pruner::TransactionInfoPruner, transaction_pruner::TransactionPruner,
+            write_set_pruner::WriteSetPruner,
         },
     },
-    schema::{
-        db_metadata::{DbMetadataKey, DbMetadataValue},
-        transaction::TransactionSchema,
-    },
     EventStore, TransactionStore,
 };
-use aptos_logger::warn;
-use aptos_schemadb::{ReadOptions, SchemaBatch, DB};
+use anyhow::Result;
 use aptos_types::transaction::{AtomicVersion, Version};
-use std::sync::{atomic::Ordering, Arc};
+use std::{
+    cmp::min,
+    sync::{atomic::Ordering, Arc},
+};
 
 pub const LEDGER_PRUNER_NAME: &str = "ledger_pruner";
 
 /// Responsible for pruning everything except for the state tree.
 pub(crate) struct LedgerPruner {
-    db: Arc<DB>,
     /// Keeps track of the target version that the pruner needs to achieve.
     target_version: AtomicVersion,
-    min_readable_version: AtomicVersion,
-    transaction_store_pruner: Arc<dyn DBSubPruner + Send + Sync>,
-    version_data_pruner: Arc<dyn DBSubPruner + Send + Sync>,
-    event_store_pruner: Arc<dyn DBSubPruner + Send + Sync>,
-    write_set_pruner: Arc<dyn DBSubPruner + Send + Sync>,
+
+    progress: AtomicVersion,
```

Comment:
what's wrong with `min_readable_version`?

Reply:
ditto, the `pending` version is theoretically not readable.